src/foam/core/XMLSupport.java

```diff
@@ -0,0 +1,199 @@
+package foam.core;
+
+import com.sun.xml.internal.txw2.output.IndentingXMLStreamWriter;
+import foam.dao.ListSink;
+import foam.dao.XMLDAO;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import javax.xml.stream.*;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+
+public class XMLSupport {
+
+  // TODO: Planning to add support for other date formats in the future
+
+  public static List<FObject> fromXML(X x, XMLStreamReader xmlr) {
+    List<FObject> objList = new ArrayList<FObject>();
+    try {
+      int eventType;
+      Object clsInstance = null;
+      while ( xmlr.hasNext() ) {
+        eventType = xmlr.next();
+        switch ( eventType ) {
+          case XMLStreamConstants.START_DOCUMENT:
+            break;
+          case XMLStreamConstants.START_ELEMENT:
+            if ( xmlr.getLocalName().equals("object") ) {
+              // Create new fObject
+              String objClass = xmlr.getAttributeValue(null, "class");
+              Class cls;
+
+              try {
+                cls = Class.forName(objClass);
+                clsInstance = x.create(cls);
+              } catch (ClassNotFoundException ex) {
+
+              }
+              // Object properties
+              copyFromXML((FObject) clsInstance, xmlr);
+              if ( clsInstance != null ) {
+                objList.add((FObject) clsInstance);
+              }
+            }
+            break;
+          case XMLStreamConstants.END_ELEMENT:
+            break;
+          case XMLStreamConstants.END_DOCUMENT:
+            break;
+        }
+      }
+      xmlr.close();
+    } catch (XMLStreamException ex) {
+    }
+    return objList;
+  }
+
+  public static List<FObject> fromXML(X x, String fileName) throws IOException {
+    XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+    XMLStreamReader xmlr = null;
+    try {
+      xmlr = xmlInputFactory.createXMLStreamReader(new FileReader(fileName));
+    } catch (IOException ex) {
+      throw new IOException("Could not create/file with given fileName");
+    } catch (XMLStreamException ex) {
+
+    }
+    return fromXML(x, xmlr);
+  }
+
+  public static void copyFromXML(FObject obj, XMLStreamReader reader) throws XMLStreamException {
+    try {
+      PropertyInfo prop = null;
+      while ( reader.hasNext() ) {
+        int eventType;
+        eventType = reader.next();
+        switch ( eventType ) {
+          case XMLStreamConstants.START_ELEMENT:
+            ClassInfo cInfo = obj.getClassInfo();
+            prop = (PropertyInfo) cInfo.getAxiomByName(reader.getLocalName());
+            if ( prop == null ) {
```

Comment:
Not an error, otherwise it will prevent us from loading XML files from older versions of the model. At most, log a warning.